gqlalchemy/transformations/translators/pyg_translator.py

```diff
@@ -0,0 +1,183 @@
+# Copyright (c) 2016-2022 Memgraph Ltd. [https://memgraph.com]
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from torch_geometric.data import HeteroData, Data
+import torch
+
+from gqlalchemy.transformations.translators.translator import Translator
+from gqlalchemy.transformations.constants import EDGE_INDEX, PYG_ID, NUM_NODES
+from gqlalchemy.memgraph_constants import MG_HOST, MG_PORT, MG_USERNAME, MG_PASSWORD, MG_ENCRYPTED, MG_CLIENT_NAME, MG_LAZY
+
+class PyGTranslator(Translator):
+    
+    def __init__(self, default_node_label="NODE", default_edge_type="RELATIONSHIP",
+        host: str = MG_HOST,
+        port: int = MG_PORT,
+        username: str = MG_USERNAME,
+        password: str = MG_PASSWORD,
+        encrypted: bool = MG_ENCRYPTED,
+        client_name: str = MG_CLIENT_NAME,
+        lazy: bool = MG_LAZY,) -> None:
+        super().__init__(default_node_label, default_edge_type, host, port, username, password, encrypted, client_name, lazy)
+
+    @classmethod
+    def get_node_properties(cls, graph, node_label: str, node_id: int):
+        """Extracts node properties from heterogeneous graph based on the node_label.
+        Args:
+            graph: A reference to the PyG graph.
+            node_label: Node label
+            node_id: Node_id
+        """
+        node_properties = {}
+        for iter_node_label, iter_node_properties in graph.node_items():
+            if iter_node_label == node_label:
+                for property_name, property_values in iter_node_properties.items():
+                    if property_name != NUM_NODES:
+                        node_properties[property_name] = property_values[node_id]
```

Comment:
I think you can reduce indenting here:

`if iter_node_label != node_label:
    continue
for property_name, property_values in iter_node_properties.items():
    if property_name == NUM_NODES:
       continue
    node_properties[property_name] = property_values[node_id]`

Reply:
Yep